Helper/Entity/ProductHelper.php

```diff
@@ -1163,10 +1163,15 @@ public function canProductBeReindexed($product, $storeId, $isChildProduct = fals
         return true;
     }
 
+    /**
+     * @param Product $product
+     * @param int $storeId
+     * @return bool|int
+     */
     public function productIsInStock($product, $storeId)
     {
         $stockItem = $this->stockRegistry->getStockItem($product->getId());
 
-        return $stockItem->getIsInStock();
+        return ($product->isSaleable() && $stockItem->getIsInStock());
```

Comment:
```suggestion
        return $product->isSaleable() && $stockItem->getIsInStock();
```

Reply:
@damcou , thank you for review!

Brackets have been removed.
Branch develop-1.x has been merged.

Reply:
Sorry, I missed the PHPdoc issue ;)